File: login_test.js
const { Builder, By } = require('selenium-webdriver');
require('chromedriver');
const assert = require('assert');

describe('SauceDemo UI Automation', function () {
  let driver;

  before(async function () {
    driver = await new Builder().forBrowser('chrome').build();
    await driver.get('https://www.saucedemo.com/');
   
    await driver.findElement(By.id('user-name')).sendKeys('standard_user');
    await driver.findElement(By.id('password')).sendKeys('secret_sauce');
    await driver.findElement(By.id('login-button')).click();
    await driver.sleep(2000);
  });

  it('should sort products from A to Z', async function () {
    const sortElement = await driver.findElement(By.className('product_sort_container'));
    await sortElement.sendKeys('Name (A to Z)');
    const value = await sortElement.getAttribute('value');
    console.log('Sort value:', value);
    assert.equal(value, 'az');
  });

  after(async function () {
    await driver.quit();
  });
});

File: test/login_test.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import Select
import time

# Setup browser
driver = webdriver.Chrome()
driver.get("https://www.saucedemo.com/")

# Login
driver.find_element(By.ID, "user-name").send_keys("standard_user")
driver.find_element(By.ID, "password").send_keys("secret_sauce")
driver.find_element(By.ID, "login-button").click()

time.sleep(2)  # Tunggu halaman selesai load

# Sort produk dari A-Z
sort_dropdown = Select(driver.find_element(By.CLASS_NAME, "product_sort_container"))
sort_dropdown.select_by_visible_text("Name (A to Z)")

time.sleep(2)  # Tunggu sorting selesai

# Tutup browser
driver.quit()